src/sdk.tsx

```diff
@@ -25,15 +28,51 @@ import('./formio-init');
 
 const VERSION = getVersion();
 
+export interface SDKOptions {
+  baseUrl: string;
+  formId: string; // UUID or slug
+  basePath?: string;
+  CSPNonce?: string;
+  lang?: SupportedLocales;
+  sentryDSN?: string;
+  sentryEnv?: string;
+  languageSelectorTarget?: string | HTMLElement | null;
+  useHashRouting?: boolean;
+  onLanguageChange?: (
+    newLanguagecode: SupportedLocales,
+    initialDataReference: string | null
+  ) => void;
+}
+
 class OpenForm {
-  constructor(targetNode, opts) {
+  protected targetNode: HTMLElement;
+  protected baseUrl: string;
+  protected apiUrl: string;
+  protected initialDataReference: string | null;
+
+  protected lang: SupportedLocales;
+  protected languageSelectorTarget: HTMLElement | null;
+  protected onLanguageChange: SDKOptions['onLanguageChange'];
+
+  // resolved form definition from backend
+  protected formObject: Form | null;
+
+  protected useHashRouting: boolean;
+  protected routerBasePath: string;
+  protected browserBasePath: string;
+  protected clientBaseUrl: string;
+  protected baseDocumentTitle: string;
+
+  private root: Root | null;
```

Comment:
I opted to make most of the API protected/private/internal here because it *is* our public API that we expose to embedders, but that requires some error suppression in `sdk.spec.ts` :/